packages/slo-stat-bot/src/slo-stat-label.ts

```diff
@@ -0,0 +1,183 @@
+      const repo = context.payload.repository.name;
+      const pull_number =  context.payload.number;
+
+      let fileResponse: Response<PullsListFilesResponse> | undefined;
+      fileResponse = await handler.listFiles(context.github, owner, repo, pull_number, 100); 
+      let fileList: PullsListFilesResponseItem[] | undefined;
+
+      if (fileResponse !== undefined) fileList = fileResponse.data;
+
+      for (let i = 0; fileList !== undefined && fileList[i] !== undefined; i++) {
+        const file = fileList[i];
+        
+        if(file.filename === 'issue_slo_rules.json') {       
+          await handler.handle_slos(context, owner, repo, pull_number, file.sha);
+          break;
+        }
+      } 
+  });
+};
+
+handler.handle_slos = async function handle_slos(
+  context: Context,
+  owner: string,
+  repo: string,
+  issue_number: number,
+  file_sha: string
+) {
+    const sloString = await handler.getFileContents(context.github, owner, repo, file_sha);
+
+    if(sloString !== undefined) {
+        var sloData = JSON.parse(sloString)
+        const res: ValidationResults = await handler.lint(schema, sloData);
+        
+        await handler.commentPR(context.github, owner, repo, issue_number, res.isValid);
+        await handler.createCheck(context, res);
+    }
+}
+
+handler.getFileContents = async function getFileContents(
+  github: GitHubAPI,
+  owner: string,
+  repo: string,
+  file_sha: string,
+) {
+    try {
+      const blob = await github.git.getBlob({
+        owner,
+        repo,
+        file_sha
+      });
+      const fileContent = Buffer.from(blob.data.content, 'base64').toString('utf8');
+      return fileContent;
+    } catch (err) {
+      console.log(err);
+      return;
+    }
+}
+
+handler.listFiles = async function listFiles(
+  github: GitHubAPI,
+  owner: string,
+  repo: string,
+  pull_number: number,
+  per_page: number
+) {
+  try {
+    const listOfFiles = await github.pulls.listFiles({
+      owner,
+      repo,
+      pull_number,
+      per_page
+    });
+    return listOfFiles;
+  } catch (err) {
+    console.log(err);
+    return;
+  }
+};
+
+handler.lint = async function lint (
+  schema: JSON, 
+  sloData: JSON
+) {
+    var ajv = new Ajv();
+    var validate = await ajv.compile(schema);
+    var isValid = await validate(sloData);
+
+    const res: ValidationResults = {
+      isValid: isValid,
+      errors: validate.errors
+    }
+    return res;
+}
+
+handler.commentPR = async function commentPR(
+  github: GitHubAPI,
+  owner: string, 
+  repo: string,
+  issue_number: number,
+  isValid: Promise<boolean>
+) {
+  let body: string = "";
+
+  if(!isValid) {
+    body = 'ERROR: "issue_slo_rules.json" file is not valid with Json schema'
+    try {
+      await github.issues.createComment({
+        owner,
+        repo,
+        issue_number,
+        body
+      });
+    } catch (err) {
+        console.log(err);
+        return;
+    }
+  }
+  return;
+}
```

Comment:
For this function, I'd consider either moving the declaration of the `body` variable into the `if(!isValid)` block so as to not "waste" an allocation on the stack.

That said, if we include a second check and comment if it IS valid (i.e. the discussed "Here are the changes to your SLO rules" comment), then leaving that declaration is perfectly fine.

Reply:
I can see from a reviewers pov how including the specific changes of SLO rules can make it easier for them. I will include this feature in the next iteraton.